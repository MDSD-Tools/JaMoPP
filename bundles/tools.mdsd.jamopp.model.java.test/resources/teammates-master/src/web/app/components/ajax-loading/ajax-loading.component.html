<div class="loading-container">
  <div class="spinner-border spinner-border-sm text-{{useBlueSpinner ? 'primary' : color}}" role="status"></div>
</div>
