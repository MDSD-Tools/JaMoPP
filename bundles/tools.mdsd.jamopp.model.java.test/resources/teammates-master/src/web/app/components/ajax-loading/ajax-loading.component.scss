.loading-container {
  display: inline;
  margin-right: 10px;
}
