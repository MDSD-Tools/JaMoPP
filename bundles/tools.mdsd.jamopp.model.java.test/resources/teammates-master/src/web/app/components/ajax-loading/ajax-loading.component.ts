import { Component, Input, OnInit } from '@angular/core';

/**
 * Displaying the ajax loader.
 */
@Component({
  selector: 'tm-ajax-loading',
  templateUrl: './ajax-loading.component.html',
  styleUrls: ['./ajax-loading.component.scss'],
})
export class AjaxLoadingComponent implements OnInit {

  @Input()
  useBlueSpinner: boolean = false;

  @Input()
  color: string = 'white';

  constructor() { }

  ngOnInit(): void {
  }

}
