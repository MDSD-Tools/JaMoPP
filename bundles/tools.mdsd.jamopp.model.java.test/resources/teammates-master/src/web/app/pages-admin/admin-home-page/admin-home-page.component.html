<div class="card bg-light top-padded">
  <div class="card-body">
    <div>
      <strong>Adding Multiple Instructors</strong>
      <p class="text-muted">
        <i class="fas fa-exclamation-circle"></i> Add Instructor Details in the format: Name | Email | Institution
      </p>
      <textarea class="form-control" id="instructor-details-single-line" [(ngModel)]="instructorDetails"></textarea>
    </div>

    <div class="top-padded">
      <button class="btn btn-primary" id="add-instructor-single-line" (click)="validateAndAddInstructorDetails()">Add Instructors</button>
    </div>
  </div>
</div>

<div class="card bg-light top-padded">
  <div class="card-body">
    <div>
      <strong>Adding a Single Instructor</strong>
    </div>

    <div class="top-padded">
      <strong>Name:</strong>
      <input class="form-control" type="text" id="instructor-name" [(ngModel)]="instructorName">
    </div>

    <div class="top-padded">
      <strong>Email: </strong>
      <input class="form-control" type="text" id="instructor-email" [(ngModel)]="instructorEmail">
    </div>

    <div class="top-padded">
      <strong>Institution: </strong>
      <input class="form-control" type="text" id="instructor-institution" [(ngModel)]="instructorInstitution">
    </div>

    <div class="top-padded">
      <button class="btn btn-primary" id="add-instructor" (click)="validateAndAddInstructorDetail()">Add Instructor</button>
    </div>
  </div>
</div>

<div class="card bg-light top-padded" *ngIf="instructorsConsolidated.length">
  <div class="card-header bg-primary text-white">
    <strong>Result</strong>
  </div>
  <div class="card-body">
    <div class="table-responsive">
      <table class="table table-striped table-hover">
        <thead>
          <tr>
            <th>Name</th>
            <th>Email</th>
            <th>Institution</th>
            <th>Action</th>
            <th>Status</th>
            <th>Message</th>
          </tr>
        </thead>
        <tbody>
          <tr *ngFor="let instructor of instructorsConsolidated; let i = index">
            <td>{{ instructor.name }}</td>
            <td>{{ instructor.email }}</td>
            <td>{{ instructor.institution }}</td>
            <td>
              <div class="button-group" *ngIf="instructor.status === 'PENDING' || instructor.status === 'FAIL'">
                <button class="action-btn btn btn-sm btn-primary text-nowrap" id="add-instructor-{{ i }}" style="margin-right: 10px;" (click)="addInstructor(i)" [disabled]="activeRequests > 0">Add Instructor</button>
                <button class="action-btn btn btn-sm btn-danger" id="cancel-instructor-{{ i }}" (click)="cancelInstructor(i)">Cancel</button>
              </div>
            </td>
            <td>{{ instructor.status }}</td>
            <td style="width: 500px;">
              <span *ngIf="instructor.status === 'SUCCESS'" id="message-instructor-{{ i }}">
                Instructor "{{ instructor.name }}" has been successfully created. [<a href="{{ instructor.joinLink }}">join link</a>]
              </span>
              <span class="message" *ngIf="instructor.status === 'FAIL'" id="message-instructor-{{ i }}">{{ instructor.message }}</span>
            </td>
          </tr>
        </tbody>
      </table>
      <button class="btn btn-primary top-padded" (click)="addAllInstructors()" id="add-all-instructors" [disabled]="activeRequests > 0 || isAddingInstructors">
        <tm-ajax-loading *ngIf="isAddingInstructors"></tm-ajax-loading>
        Add All Instructors
      </button>
    </div>
  </div>
</div>
