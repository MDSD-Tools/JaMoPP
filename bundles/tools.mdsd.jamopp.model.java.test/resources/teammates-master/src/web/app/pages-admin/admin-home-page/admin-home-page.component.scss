.top-padded {
  margin-top: 20px;
}

.button-group {
  display: flex;
}

.action-btn {
  margin: 2px;
}

.message {
  display: inline-block;
  width: inherit;
}
