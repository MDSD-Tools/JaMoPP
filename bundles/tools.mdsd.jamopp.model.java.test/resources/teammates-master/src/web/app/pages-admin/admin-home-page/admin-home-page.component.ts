import { Component } from '@angular/core';
import { finalize } from 'rxjs/operators';
import { AccountService } from '../../../services/account.service';
import { JoinLink } from '../../../types/api-output';
import { ErrorMessageOutput } from '../../error-message-output';

interface InstructorData {
  name: string;
  email: string;
  institution: string;
  status: string;
  joinLink?: string;
  message?: string;
}

/**
 * Admin home page.
 */
@Component({
  selector: 'tm-admin-home-page',
  templateUrl: './admin-home-page.component.html',
  styleUrls: ['./admin-home-page.component.scss'],
})
export class AdminHomePageComponent {

  instructorDetails: string = '';
  instructorName: string = '';
  instructorEmail: string = '';
  instructorInstitution: string = '';

  instructorsConsolidated: InstructorData[] = [];
  activeRequests: number = 0;

  isAddingInstructors: boolean = false;

  constructor(private accountService: AccountService) {}

  /**
   * Validates and adds the instructor details filled with first form.
   */
  validateAndAddInstructorDetails(): void {
    const invalidLines: string[] = [];
    for (const instructorDetail of this.instructorDetails.split(/\r?\n/)) {
      const instructorDetailSplit: string[] = instructorDetail.split(/[|\t]/).map((item: string) => item.trim());
      if (instructorDetailSplit.length < 3) {
        // TODO handle error
        invalidLines.push(instructorDetail);
        continue;
      }
      if (!instructorDetailSplit[0] || !instructorDetailSplit[1] || !instructorDetailSplit[2]) {
        // TODO handle error
        invalidLines.push(instructorDetail);
        continue;
      }
      this.instructorsConsolidated.push({
        name: instructorDetailSplit[0],
        email: instructorDetailSplit[1],
        institution: instructorDetailSplit[2],
        status: 'PENDING',
      });
    }
    this.instructorDetails = invalidLines.join('\r\n');
  }

  /**
   * Validates and adds the instructor detail filled with second form.
   */
  validateAndAddInstructorDetail(): void {
    if (!this.instructorName || !this.instructorEmail || !this.instructorInstitution) {
      // TODO handle error
      return;
    }
    this.instructorsConsolidated.push({
      name: this.instructorName,
      email: this.instructorEmail,
      institution: this.instructorInstitution,
      status: 'PENDING',
    });
    this.instructorName = '';
    this.instructorEmail = '';
    this.instructorInstitution = '';
  }

  /**
   * Adds the instructor at the i-th index.
   */
  addInstructor(i: number): void {
    const instructor: InstructorData = this.instructorsConsolidated[i];
    if (instructor.status !== 'PENDING' && instructor.status !== 'FAIL') {
      return;
    }
    this.activeRequests += 1;
    instructor.status = 'ADDING';

    this.isAddingInstructors = true;
    this.accountService.createAccount({
      instructorEmail: instructor.email,
      instructorName: instructor.name,
      instructorInstitution: instructor.institution,
    })
        .pipe(finalize(() => this.isAddingInstructors = false))
        .subscribe((resp: JoinLink) => {
          instructor.status = 'SUCCESS';
          instructor.joinLink = resp.joinLink;
          this.activeRequests -= 1;
        }, (resp: ErrorMessageOutput) => {
          instructor.status = 'FAIL';
          instructor.message = resp.error.message;
          this.activeRequests -= 1;
        });
  }

  /**
   * Cancels the instructor at the i-th index.
   */
  cancelInstructor(i: number): void {
    this.instructorsConsolidated.splice(i, 1);
  }

  /**
   * Adds all the pending and failed-to-add instructors.
   */
  addAllInstructors(): void {
    for (let i: number = 0; i < this.instructorsConsolidated.length; i += 1) {
      this.addInstructor(i);
    }
  }

}
