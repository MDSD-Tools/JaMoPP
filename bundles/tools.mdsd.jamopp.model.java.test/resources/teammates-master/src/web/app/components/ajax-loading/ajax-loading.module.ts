import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { AjaxLoadingComponent } from './ajax-loading.component';

/**
 * Ajax loading module.
 */
@NgModule({
  imports: [
    CommonModule,
  ],
  declarations: [
    AjaxLoadingComponent,
  ],
  exports: [
    AjaxLoadingComponent,
  ],
})
export class AjaxLoadingModule { }
