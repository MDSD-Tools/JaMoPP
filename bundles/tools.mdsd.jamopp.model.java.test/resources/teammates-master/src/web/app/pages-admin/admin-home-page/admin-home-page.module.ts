import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { RouterModule, Routes } from '@angular/router';
import { AjaxLoadingModule } from '../../components/ajax-loading/ajax-loading.module';
import { AdminHomePageComponent } from './admin-home-page.component';

const routes: Routes = [
  {
    path: '',
    component: AdminHomePageComponent,
  },
];

/**
 * Module for admin home page.
 */
@NgModule({
  declarations: [
    AdminHomePageComponent,
  ],
  exports: [
    AdminHomePageComponent,
  ],
  imports: [
    CommonModule,
    FormsModule,
    RouterModule.forChild(routes),
    AjaxLoadingModule,
  ],
})
export class AdminHomePageModule { }
