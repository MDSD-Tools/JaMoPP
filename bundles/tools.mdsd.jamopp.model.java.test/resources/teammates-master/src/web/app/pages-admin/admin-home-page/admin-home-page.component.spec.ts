import { HttpClientTestingModule } from '@angular/common/http/testing';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { FormsModule } from '@angular/forms';
import { of, throwError } from 'rxjs';
import { AccountService } from '../../../services/account.service';
import { AjaxLoadingModule } from '../../components/ajax-loading/ajax-loading.module';
import { LoadingSpinnerModule } from '../../components/loading-spinner/loading-spinner.module';
import { AdminHomePageComponent } from './admin-home-page.component';

describe('AdminHomePageComponent', () => {
  let component: AdminHomePageComponent;
  let fixture: ComponentFixture<AdminHomePageComponent>;
  let service: AccountService;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [AdminHomePageComponent],
      imports: [
        FormsModule,
        HttpClientTestingModule,
        LoadingSpinnerModule,
        AjaxLoadingModule,
      ],
      providers: [AccountService],
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(AdminHomePageComponent);
    service = TestBed.inject(AccountService);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should add one instructor to list if all fields are filled', () => {
    component.instructorName = 'Instructor Name';
    component.instructorEmail = 'instructor@example.com';
    component.instructorInstitution = 'Instructor Institution';
    fixture.detectChanges();

    const button: any = fixture.debugElement.nativeElement.querySelector('#add-instructor');
    button.click();

    expect(component.instructorName).toEqual('');
    expect(component.instructorEmail).toEqual('');
    expect(component.instructorInstitution).toEqual('');
    expect(component.instructorsConsolidated.length).toEqual(1);
    expect(component.instructorsConsolidated[0]).toEqual({
      email: 'instructor@example.com',
      institution: 'Instructor Institution',
      name: 'Instructor Name',
      status: 'PENDING',
    });
  });

  it('should not add one instructor to list if some fields are empty', () => {
    component.instructorName = 'Instructor Name';
    component.instructorEmail = '';
    component.instructorInstitution = 'Instructor Institution';
    fixture.detectChanges();

    const button: any = fixture.debugElement.nativeElement.querySelector('#add-instructor');
    button.click();

    expect(component.instructorName).toEqual('Instructor Name');
    expect(component.instructorEmail).toEqual('');
    expect(component.instructorInstitution).toEqual('Instructor Institution');
    expect(component.instructorsConsolidated.length).toEqual(0);

    component.instructorName = '';
    component.instructorEmail = 'instructor@example.com';

    button.click();

    expect(component.instructorName).toEqual('');
    expect(component.instructorEmail).toEqual('instructor@example.com');
    expect(component.instructorInstitution).toEqual('Instructor Institution');
    expect(component.instructorsConsolidated.length).toEqual(0);

    component.instructorName = 'Instructor Name';
    component.instructorInstitution = '';

    button.click();

    expect(component.instructorName).toEqual('Instructor Name');
    expect(component.instructorEmail).toEqual('instructor@example.com');
    expect(component.instructorInstitution).toEqual('');
    expect(component.instructorsConsolidated.length).toEqual(0);
  });

  it('should only add valid instructor details in the single line field', () => {
    component.instructorDetails = [
        'Instructor A | instructora@example.com | Institution A',
        'Instructor B | instructorb@example.com',
        'Instructor C | | instructorc@example.com',
        'Instructor D | instructord@example.com | Institution D',
        '| instructore@example.com | Institution E',
    ].join('\n');
    fixture.detectChanges();

    const button: any = fixture.debugElement.nativeElement.querySelector('#add-instructor-single-line');
    button.click();

    expect(component.instructorDetails).toEqual([
      'Instructor B | instructorb@example.com',
      'Instructor C | | instructorc@example.com',
      '| instructore@example.com | Institution E',
    ].join('\r\n'));
    expect(component.instructorsConsolidated.length).toEqual(2);
    expect(component.instructorsConsolidated[0]).toEqual({
      email: 'instructora@example.com',
      institution: 'Institution A',
      name: 'Instructor A',
      status: 'PENDING',
    });
    expect(component.instructorsConsolidated[1]).toEqual({
      email: 'instructord@example.com',
      institution: 'Institution D',
      name: 'Instructor D',
      status: 'PENDING',
    });
  });

  it('should remove instructor out of queue if CANCEL is requested', () => {
    component.instructorsConsolidated = [
      {
        name: 'Instructor A',
        email: 'instructora@example.com',
        institution: 'Sample Institution A',
        status: 'PENDING',
        joinLink: 'This should not be displayed',
        message: 'This should not be displayed',
      },
    ];
    fixture.detectChanges();

    const button: any = fixture.debugElement.nativeElement.querySelector('#cancel-instructor-0');
    button.click();

    expect(component.instructorsConsolidated.length).toEqual(0);
  });

  it('should add instructor and update field when successful', () => {
    component.instructorsConsolidated = [
      {
        name: 'Instructor A',
        email: 'instructora@example.com',
        institution: 'Sample Institution A',
        status: 'PENDING',
        joinLink: 'This should not be displayed',
        message: 'This should not be displayed',
      },
    ];
    spyOn(service, 'createAccount').and.returnValue(of({
      joinLink: 'http://localhost:4200/web/join',
    }));
    fixture.detectChanges();

    const button: any = fixture.debugElement.nativeElement.querySelector('#add-instructor-0');
    button.click();

    expect(component.instructorsConsolidated[0].status).toEqual('SUCCESS');
    expect(component.instructorsConsolidated[0].joinLink).toEqual('http://localhost:4200/web/join');
    expect(component.activeRequests).toEqual(0);
  });

  it('should not add instructor and update field when failure', () => {
    component.instructorsConsolidated = [
      {
        name: 'Instructor A',
        email: 'instructora@example.com',
        institution: 'Sample Institution A',
        status: 'PENDING',
        joinLink: 'This should not be displayed',
        message: 'This should not be displayed',
      },
    ];
    spyOn(service, 'createAccount').and.returnValue(throwError({
      error: {
        message: 'This is the error message',
      },
    }));
    fixture.detectChanges();

    const button: any = fixture.debugElement.nativeElement.querySelector('#add-instructor-0');
    button.click();

    expect(component.instructorsConsolidated[0].status).toEqual('FAIL');
    expect(component.instructorsConsolidated[0].message).toEqual('This is the error message');
    expect(component.activeRequests).toEqual(0);
  });

  it('should add all instructors when prompted', () => {
    component.instructorsConsolidated = [
      {
        name: 'Instructor A',
        email: 'instructora@example.com',
        institution: 'Sample Institution A',
        status: 'PENDING',
        joinLink: 'This should not be displayed',
        message: 'This should not be displayed',
      },
      {
        name: 'Instructor B',
        email: 'instructorb@example.com',
        institution: 'Sample Institution B',
        status: 'SUCCESS',
        joinLink: 'http://localhost:4200/web/join',
        message: 'This should not be displayed',
      },
      {
        name: 'Instructor C',
        email: 'instructorc@example.com',
        institution: 'Sample Institution C',
        status: 'FAIL',
        joinLink: 'This should not be displayed',
        message: 'The instructor cannot be added for some reason',
      },
    ];
    // No need to spy here as this test only tests the number of active requests added
    // Testing of adding individual instructors have been done before
    fixture.detectChanges();

    const button: any = fixture.debugElement.nativeElement.querySelector('#add-all-instructors');
    button.click();

    expect(component.instructorsConsolidated[0].status).toEqual('ADDING');
    expect(component.instructorsConsolidated[1].status).toEqual('SUCCESS');
    expect(component.instructorsConsolidated[2].status).toEqual('ADDING');
    expect(component.activeRequests).toEqual(2);
  });

  it('should snap with default view', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('should snap with some instructors details', () => {
    component.instructorsConsolidated = [
      {
        name: 'Instructor A',
        email: 'instructora@example.com',
        institution: 'Sample Institution A',
        status: 'PENDING',
        joinLink: 'This should not be displayed',
        message: 'This should not be displayed',
      },
      {
        name: 'Instructor B',
        email: 'instructorb@example.com',
        institution: 'Sample Institution B',
        status: 'SUCCESS',
        joinLink: 'http://localhost:4200/web/join',
        message: 'This should not be displayed',
      },
      {
        name: 'Instructor C',
        email: 'instructorc@example.com',
        institution: 'Sample Institution C',
        status: 'FAIL',
        joinLink: 'This should not be displayed',
        message: 'The instructor cannot be added for some reason',
      },
    ];
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should snap with disabled adding instructor button if there are active requests', () => {
    component.instructorsConsolidated = [
      {
        name: 'Instructor A',
        email: 'instructora@example.com',
        institution: 'Sample Institution A',
        status: 'ADDING',
        joinLink: 'This should not be displayed',
        message: 'This should not be displayed',
      },
      {
        name: 'Instructor B',
        email: 'instructorb@example.com',
        institution: 'Sample Institution B',
        status: 'PENDING',
        joinLink: 'This should not be displayed',
        message: 'This should not be displayed',
      },
    ];
    component.activeRequests = 1;
    component.isAddingInstructors = true;

    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should add multiple instructors split by tabs', () => {
    component.instructorDetails = `Instructor A   \t  instructora@example.com \t  Sample Institution A\n
     Instructor B \t instructorb@example.com \t Sample Institution B`;

    fixture.detectChanges();

    const button: any = fixture.debugElement.nativeElement.querySelector('#add-instructor-single-line');
    button.click();

    expect(component.instructorsConsolidated.length).toEqual(2);
    expect(component.instructorsConsolidated[0]).toEqual(
      {
        name: 'Instructor A',
        email: 'instructora@example.com',
        institution: 'Sample Institution A',
        status: 'PENDING',
      },
    );
    expect(component.instructorsConsolidated[1]).toEqual(
      {
        name: 'Instructor B',
        email: 'instructorb@example.com',
        institution: 'Sample Institution B',
        status: 'PENDING',
      },
    );
  });

  it('should add multiple instructors split by vertical bars', () => {
    component.instructorDetails = `Instructor A | instructora@example.com | Sample Institution A\n
        Instructor B | instructorb@example.com | Sample Institution B`;

    fixture.detectChanges();

    const button: any = fixture.debugElement.nativeElement.querySelector('#add-instructor-single-line');
    button.click();

    expect(component.instructorsConsolidated.length).toEqual(2);
    expect(component.instructorsConsolidated[0]).toEqual(
      {
        name: 'Instructor A',
        email: 'instructora@example.com',
        institution: 'Sample Institution A',
        status: 'PENDING',
      },
    );
    expect(component.instructorsConsolidated[1]).toEqual(
      {
        name: 'Instructor B',
        email: 'instructorb@example.com',
        institution: 'Sample Institution B',
        status: 'PENDING',
      },
    );
  });
});
