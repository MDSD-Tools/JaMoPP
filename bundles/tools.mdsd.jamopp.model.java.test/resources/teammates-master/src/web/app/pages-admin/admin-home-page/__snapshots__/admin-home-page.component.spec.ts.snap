// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminHomePageComponent should snap with default view 1`] = `
<tm-admin-home-page
  accountService={[Function AccountService]}
  activeRequests="0"
  instructorDetails=""
  instructorEmail=""
  instructorInstitution=""
  instructorName=""
  instructorsConsolidated={[Function Array]}
  isAddingInstructors="false"
>
  <div
    class="card bg-light top-padded"
  >
    <div
      class="card-body"
    >
      <div>
        <strong>
          Adding Multiple Instructors
        </strong>
        <p
          class="text-muted"
        >
          <i
            class="fas fa-exclamation-circle"
          />
           Add Instructor Details in the format: Name | Email | Institution 
        </p>
        <textarea
          class="form-control ng-untouched ng-pristine ng-valid"
          id="instructor-details-single-line"
        />
      </div>
      <div
        class="top-padded"
      >
        <button
          class="btn btn-primary"
          id="add-instructor-single-line"
        >
          Add Instructors
        </button>
      </div>
    </div>
  </div><div
    class="card bg-light top-padded"
  >
    <div
      class="card-body"
    >
      <div>
        <strong>
          Adding a Single Instructor
        </strong>
      </div>
      <div
        class="top-padded"
      >
        <strong>
          Name:
        </strong>
        <input
          class="form-control ng-untouched ng-pristine ng-valid"
          id="instructor-name"
          type="text"
        />
      </div>
      <div
        class="top-padded"
      >
        <strong>
          Email: 
        </strong>
        <input
          class="form-control ng-untouched ng-pristine ng-valid"
          id="instructor-email"
          type="text"
        />
      </div>
      <div
        class="top-padded"
      >
        <strong>
          Institution: 
        </strong>
        <input
          class="form-control ng-untouched ng-pristine ng-valid"
          id="instructor-institution"
          type="text"
        />
      </div>
      <div
        class="top-padded"
      >
        <button
          class="btn btn-primary"
          id="add-instructor"
        >
          Add Instructor
        </button>
      </div>
    </div>
  </div>
</tm-admin-home-page>
`;

exports[`AdminHomePageComponent should snap with disabled adding instructor button if there are active requests 1`] = `
<tm-admin-home-page
  accountService={[Function AccountService]}
  activeRequests={[Function Number]}
  instructorDetails=""
  instructorEmail=""
  instructorInstitution=""
  instructorName=""
  instructorsConsolidated={[Function Array]}
  isAddingInstructors={[Function Boolean]}
>
  <div
    class="card bg-light top-padded"
  >
    <div
      class="card-body"
    >
      <div>
        <strong>
          Adding Multiple Instructors
        </strong>
        <p
          class="text-muted"
        >
          <i
            class="fas fa-exclamation-circle"
          />
           Add Instructor Details in the format: Name | Email | Institution 
        </p>
        <textarea
          class="form-control ng-untouched ng-pristine ng-valid"
          id="instructor-details-single-line"
        />
      </div>
      <div
        class="top-padded"
      >
        <button
          class="btn btn-primary"
          id="add-instructor-single-line"
        >
          Add Instructors
        </button>
      </div>
    </div>
  </div><div
    class="card bg-light top-padded"
  >
    <div
      class="card-body"
    >
      <div>
        <strong>
          Adding a Single Instructor
        </strong>
      </div>
      <div
        class="top-padded"
      >
        <strong>
          Name:
        </strong>
        <input
          class="form-control ng-untouched ng-pristine ng-valid"
          id="instructor-name"
          type="text"
        />
      </div>
      <div
        class="top-padded"
      >
        <strong>
          Email: 
        </strong>
        <input
          class="form-control ng-untouched ng-pristine ng-valid"
          id="instructor-email"
          type="text"
        />
      </div>
      <div
        class="top-padded"
      >
        <strong>
          Institution: 
        </strong>
        <input
          class="form-control ng-untouched ng-pristine ng-valid"
          id="instructor-institution"
          type="text"
        />
      </div>
      <div
        class="top-padded"
      >
        <button
          class="btn btn-primary"
          id="add-instructor"
        >
          Add Instructor
        </button>
      </div>
    </div>
  </div><div
    class="card bg-light top-padded"
  >
    <div
      class="card-header bg-primary text-white"
    >
      <strong>
        Result
      </strong>
    </div>
    <div
      class="card-body"
    >
      <div
        class="table-responsive"
      >
        <table
          class="table table-striped table-hover"
        >
          <thead>
            <tr>
              <th>
                Name
              </th>
              <th>
                Email
              </th>
              <th>
                Institution
              </th>
              <th>
                Action
              </th>
              <th>
                Status
              </th>
              <th>
                Message
              </th>
            </tr>
          </thead>
          <tbody>
            
            <tr>
              <td>
                Instructor A
              </td>
              <td>
                instructora@example.com
              </td>
              <td>
                Sample Institution A
              </td>
              <td>
                
              </td>
              <td>
                ADDING
              </td>
              <td
                style="width: 500px;"
              >
                
                
              </td>
            </tr>
            <tr>
              <td>
                Instructor B
              </td>
              <td>
                instructorb@example.com
              </td>
              <td>
                Sample Institution B
              </td>
              <td>
                
                <div
                  class="button-group"
                >
                  <button
                    class="action-btn btn btn-sm btn-primary text-nowrap"
                    disabled=""
                    id="add-instructor-1"
                    style="margin-right: 10px;"
                  >
                    Add Instructor
                  </button>
                  <button
                    class="action-btn btn btn-sm btn-danger"
                    id="cancel-instructor-1"
                  >
                    Cancel
                  </button>
                </div>
              </td>
              <td>
                PENDING
              </td>
              <td
                style="width: 500px;"
              >
                
                
              </td>
            </tr>
          </tbody>
        </table>
        <button
          class="btn btn-primary top-padded"
          disabled=""
          id="add-all-instructors"
        >
          
          <tm-ajax-loading>
            <div
              class="loading-container"
            >
              <div
                class="spinner-border spinner-border-sm text-white"
                role="status"
              />
            </div>
          </tm-ajax-loading>
           Add All Instructors 
        </button>
      </div>
    </div>
  </div>
</tm-admin-home-page>
`;

exports[`AdminHomePageComponent should snap with some instructors details 1`] = `
<tm-admin-home-page
  accountService={[Function AccountService]}
  activeRequests="0"
  instructorDetails=""
  instructorEmail=""
  instructorInstitution=""
  instructorName=""
  instructorsConsolidated={[Function Array]}
  isAddingInstructors="false"
>
  <div
    class="card bg-light top-padded"
  >
    <div
      class="card-body"
    >
      <div>
        <strong>
          Adding Multiple Instructors
        </strong>
        <p
          class="text-muted"
        >
          <i
            class="fas fa-exclamation-circle"
          />
           Add Instructor Details in the format: Name | Email | Institution 
        </p>
        <textarea
          class="form-control ng-untouched ng-pristine ng-valid"
          id="instructor-details-single-line"
        />
      </div>
      <div
        class="top-padded"
      >
        <button
          class="btn btn-primary"
          id="add-instructor-single-line"
        >
          Add Instructors
        </button>
      </div>
    </div>
  </div><div
    class="card bg-light top-padded"
  >
    <div
      class="card-body"
    >
      <div>
        <strong>
          Adding a Single Instructor
        </strong>
      </div>
      <div
        class="top-padded"
      >
        <strong>
          Name:
        </strong>
        <input
          class="form-control ng-untouched ng-pristine ng-valid"
          id="instructor-name"
          type="text"
        />
      </div>
      <div
        class="top-padded"
      >
        <strong>
          Email: 
        </strong>
        <input
          class="form-control ng-untouched ng-pristine ng-valid"
          id="instructor-email"
          type="text"
        />
      </div>
      <div
        class="top-padded"
      >
        <strong>
          Institution: 
        </strong>
        <input
          class="form-control ng-untouched ng-pristine ng-valid"
          id="instructor-institution"
          type="text"
        />
      </div>
      <div
        class="top-padded"
      >
        <button
          class="btn btn-primary"
          id="add-instructor"
        >
          Add Instructor
        </button>
      </div>
    </div>
  </div><div
    class="card bg-light top-padded"
  >
    <div
      class="card-header bg-primary text-white"
    >
      <strong>
        Result
      </strong>
    </div>
    <div
      class="card-body"
    >
      <div
        class="table-responsive"
      >
        <table
          class="table table-striped table-hover"
        >
          <thead>
            <tr>
              <th>
                Name
              </th>
              <th>
                Email
              </th>
              <th>
                Institution
              </th>
              <th>
                Action
              </th>
              <th>
                Status
              </th>
              <th>
                Message
              </th>
            </tr>
          </thead>
          <tbody>
            
            <tr>
              <td>
                Instructor A
              </td>
              <td>
                instructora@example.com
              </td>
              <td>
                Sample Institution A
              </td>
              <td>
                
                <div
                  class="button-group"
                >
                  <button
                    class="action-btn btn btn-sm btn-primary text-nowrap"
                    id="add-instructor-0"
                    style="margin-right: 10px;"
                  >
                    Add Instructor
                  </button>
                  <button
                    class="action-btn btn btn-sm btn-danger"
                    id="cancel-instructor-0"
                  >
                    Cancel
                  </button>
                </div>
              </td>
              <td>
                PENDING
              </td>
              <td
                style="width: 500px;"
              >
                
                
              </td>
            </tr>
            <tr>
              <td>
                Instructor B
              </td>
              <td>
                instructorb@example.com
              </td>
              <td>
                Sample Institution B
              </td>
              <td>
                
              </td>
              <td>
                SUCCESS
              </td>
              <td
                style="width: 500px;"
              >
                
                <span
                  id="message-instructor-1"
                >
                   Instructor "Instructor B" has been successfully created. [
                  <a
                    href="http://localhost:4200/web/join"
                  >
                    join link
                  </a>
                  ] 
                </span>
                
              </td>
            </tr>
            <tr>
              <td>
                Instructor C
              </td>
              <td>
                instructorc@example.com
              </td>
              <td>
                Sample Institution C
              </td>
              <td>
                
                <div
                  class="button-group"
                >
                  <button
                    class="action-btn btn btn-sm btn-primary text-nowrap"
                    id="add-instructor-2"
                    style="margin-right: 10px;"
                  >
                    Add Instructor
                  </button>
                  <button
                    class="action-btn btn btn-sm btn-danger"
                    id="cancel-instructor-2"
                  >
                    Cancel
                  </button>
                </div>
              </td>
              <td>
                FAIL
              </td>
              <td
                style="width: 500px;"
              >
                
                
                <span
                  class="message"
                  id="message-instructor-2"
                >
                  The instructor cannot be added for some reason
                </span>
              </td>
            </tr>
          </tbody>
        </table>
        <button
          class="btn btn-primary top-padded"
          id="add-all-instructors"
        >
          
           Add All Instructors 
        </button>
      </div>
    </div>
  </div>
</tm-admin-home-page>
`;
